### PR TITLE
Adding Reflection changes back after merge from master

### PR DESCRIPTION
**Pull request by joperezr**
This adds back #11099 which was reverted by #11355 in order to be able to finish the merge from master to dev/api. 

cc: @sepidehMS @danmosemsft @weshaggard @stephentoub 

I only had a few of merge conflicts mainly in the csprojs given that master removed `<AssemblyVersion>` and `<PackageTargetFramework>` from most of the projects, so I just had to fix those plus updating the new dir.props for each library.

FYI: @kouvel @steveharter these are more or less the changes that you will need for the conflicts in your PRs.

### CODE REVIEW

File: src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.10.0</AssemblyVersion>
     <OutputType>Library</OutputType>
-    <NuGetTargetMoniker>.NETStandard,Version=v1.3</NuGetTargetMoniker>
+    <NuGetTargetMoniker>.NETStandard,Version=v1.7</NuGetTargetMoniker>
+    <PackageTargetFramework>netstandard1.7;netcore50</PackageTargetFramework>
   </PropertyGroup>
   <ItemGroup>
+    <SuppressPackageTargetFrameworkCompatibility Include="netcore50" />
```
Comment (weshaggard) [inline src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj]: Is this right? I wouldn't expect us to be updating the reference for netcore50.

File: src/System.Threading.Tasks/src/System.Threading.Tasks.csproj [hunk 2]
```diff
@@ -5,22 +5,23 @@
     <ProjectGuid>{3BCAEAA6-3A29-49EC-B334-6E7BE8BE9ABA}</ProjectGuid>
     <AssemblyName>System.Threading.Tasks</AssemblyName>
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
-    <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
+    <NuGetTargetMoniker Condition="'$(NuGetTargetMoniker)' == ''">.NETStandard,Version=v1.7</NuGetTargetMoniker>
+    <PackageTargetFramework Condition="'$(TargetGroup)'==''">netstandard1.7;netcore50</PackageTargetFramework>
```
Comment (weshaggard) [inline src/System.Threading.Tasks/src/System.Threading.Tasks.csproj]: Same question as with the ref. Why do we need the netcore50 version?

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test please (a System.Net.Security test failed with the message:Authentication failed because the remote party has closed the transport stream.)

**Comment by joperezr** [comment]
test Innerloop Windows_NT Debug Build and Test please

File: src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj (hunk 1 above)
Comment (joperezr) [inline src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj]: Also seemed wierd to me, but this is how @sepidehMS merged her PR initially so I thought there was a reason behind it. @sepidehMS any special reason why we did this ?

Comment (kouvel) [inline src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj]: I'm removing netcore50-related stuff for this project in my PR, so this should go away once I merge with it as part of my change

Comment (joperezr) [inline src/System.Threading.Tasks/ref/System.Threading.Tasks.csproj]: ok, I talked to @sepidehMS offline and it looks like this was actually added by mistake, so I'll remove my supressions.

**Comment by joperezr** [comment]
@dotnet-bot test nnerloop OSX Debug Build and Test please

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test please

**Comment by joperezr** [comment]
Any additional comments or is it ok to merge this?

**Comment by stephentoub** [comment]
LGTM, but it needs to be rebased.

**Pull request merged**